### PR TITLE
docs[DST-572]: Form field guidelines

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**
# Description
Our form field components share many common features, such as states (disabled, error, readonly) and functionality (label, control, help text). When we revise every single page, this information would be duplicated across multiple component pages, leading to redundancy and potential inconsistencies.

As we are in the process of revising our component pages, now is the perfect time to implement this solution. Establishing a centralized page will streamline future revisions, ensuring consistency and saving time as we update and maintain our documentation.

# What should be tested?
- Should explain the anatomy of form fields (label, helptext and a “control” that depends on the field e.g. input/select/radio/…)
- Should explain how validation works and what options there are (might be possible to just move the current “validation” page into this one and delete the “validation” page)
- Should explain common use cases (when to use a label, when it is okay not to, when to use a placeholder, …)

# Reviewers:

@marigold-ui/developer

### CODE REVIEW

**Comment by sarahgm** [comment]
can you edit the title of the PR so its containing the issue and the type of change? 
Also this review can be done by designer and katharina right?

File: .changeset/sweet-pumpkins-cover.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@marigold/components': patch
+'@marigold/docs': patch
+---
+
+Adding a general form guideline in docs explaining different concepts such as when to use lable and help text
```
Comment (sarahgm) [inline .changeset/sweet-pumpkins-cover.md:6]: title is missing See: confluence changeset page

**Comment by sarahgm** [comment]
I reviewed. Someone else maybe should also: @sebald @marcelkoehler @tirado-rx @KadaStrophe ?

**Review by aromko — DISMISSED** [review]

File: docs/content/foundations/form-fields/form-fields-required-state.demo.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Select } from '@marigold/components';
+
+export default () => (
+  <Select
+    label="Genre"
+    placeholder="Select genre"
+    description="Select the genre you want."
```
Comment (sarahgm) [inline docs/content/foundations/form-fields/form-fields-required-state.demo.tsx:7]:
```suggestion
```

**Review by sebald — APPROVED** [review]

**Pull request merged**